### PR TITLE
added copy feature

### PR DESCRIPTION
**Pull request by vapaavetehinen**

### CODE REVIEW

File: src/App.svelte [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
   export let servers = exampleData;
   console.log(exampleData);
+  export function grabip(text) {
+    let textbox = document.createElement('textarea');
+    textbox.value= text['ip'];
+    document.body.append(textbox);
+    textbox.select();
+    document.execCommand("copy");
+    textbox.remove();
+  }
```
Comment (akhilcacharya) [inline src/App.svelte]: Why are you passing it as an object and then indexing into the object?

Comment (akhilcacharya) [reply src/App.svelte]: Also can you verify this works across browsers/recent versions? If it doesn't we could just use navigator+polyfill ```await navigator.clipboard.writeText(ip);```

File: src/App.svelte [hunk 2]
```diff
@@ -2,6 +2,14 @@
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
   console.log(exampleData);
+  export function grabip(text) {
+    let textbox = document.createElement('textarea');
+    textbox.value= text['ip'];
```
Comment (ming08108) [inline src/App.svelte]: Add a space between value and =

You should probably setup auto formatting.

File: src/App.svelte [hunk 3]
```diff
@@ -2,6 +2,14 @@
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
   console.log(exampleData);
+  export function grabip(text) {
+    let textbox = document.createElement('textarea');
```
Comment (ming08108) [inline src/App.svelte]: This could probably be const.

File: src/App.svelte [hunk 4]
```diff
@@ -2,6 +2,14 @@
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
   console.log(exampleData);
+  export function grabip(text) {
```
Comment (peakay) [inline src/App.svelte]: are we gonna do snake case camel case or no case

File: src/App.svelte (hunk 2 above)
Comment (peakay) [inline src/App.svelte]: inconsistent space

File: src/App.svelte (hunk 1 above)
Comment (ming08108) [reply src/App.svelte]: Use modern api's + polyfills when possible

**Review by peakay — CHANGES_REQUESTED** [review]

**Review by ming08108 — CHANGES_REQUESTED** [review]

File: src/App.svelte (hunk 4 above)
Comment (ming08108) [reply src/App.svelte]: Agreed we should be doing camelCase

File: src/App.svelte (hunk 1 above)
Comment (akhilcacharya) [reply src/App.svelte]: Also, design thing - we might want a visual indicator that its been copied. Better for discoverability

File: src/App.svelte (hunk 4 above)
Comment (akhilcacharya) [reply src/App.svelte]: camelCase for JS

File: src/App.svelte [hunk 5]
```diff
@@ -2,6 +2,14 @@
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
   console.log(exampleData);
```
Comment (akhilcacharya) [inline src/App.svelte]: Remove unnecessary console log

Comment (vapaavetehinen) [reply src/App.svelte]: that one's not my fault, but I'll do it

Comment (peakay) [reply src/App.svelte]: don't do it, let's not do this "while we're here" garbage

Comment (peakay) [reply src/App.svelte]: that's operationally risky

Comment (akhilcacharya) [reply src/App.svelte]: **its right there**

File: src/App.svelte (hunk 1 above)
Comment (vapaavetehinen) [reply src/App.svelte]: pretty sure this code has greater compatibility across browsers than writeText

Comment (vapaavetehinen) [reply src/App.svelte]: @ming08108 said he wanted to get rid of libraries

File: src/App.svelte (hunk 5 above)
Comment (ming08108) [reply src/App.svelte]: but its free real estate

File: src/App.svelte [hunk 6]
```diff
@@ -27,7 +29,7 @@
   {#each servers as { name, ip, banner }}
     <div class="server">
       <h2>{name}</h2>
-      <h2 class="ip">{ip}</h2>
+      <h2 class="ip">{ip}<button on:click={() => grabIp({ip}['ip'])}>Copy</button></h2>
```
Comment (akhilcacharya) [inline src/App.svelte]: It's...its literally just ``grabIp(ip)``

Comment (akhilcacharya) [reply src/App.svelte]: ^

File: src/App.svelte [hunk 7]
```diff
@@ -1,7 +1,9 @@
 <script>
   import { exampleData } from "./ExampleData.js";
   export let servers = exampleData;
-  console.log(exampleData);
+  export async function grabIp(text) {
```
Comment (ming08108) [inline src/App.svelte]: this grabIP function is unnecessary since all it does is call writeText. Just pass in navigator.clipboard.writeText to on:click.

**Review by ming08108 — CHANGES_REQUESTED** [review]

**Review by ming08108 — APPROVED** [review]
Nice. It is a one liner now.

Should figure out the webpack polyfill thing at some point for older browsers.

**Pull request merged**